pkg/automationconfig/automation_config_builder.go

```diff
@@ -50,13 +53,23 @@ func (b *Builder) SetName(name string) *Builder {
 	return b
 }
 
+func (b *Builder) AddVersion(version MongoDbVersionConfig) *Builder {
+	for idx, _ := range version.Builds {
+		if version.Builds[idx].Modules == nil {
+			version.Builds[idx].Modules = make([]string, 0)
+		}
+	}
+	b.versions = append(b.versions, version)
```

Comment:
this needs to be inside the loop

Reply:
The automation config has many `MongoDbVersionConfig`s, which in turn, have multiple `Builds`.

This function only receives 1 `MongoDbVersionConfig`, that need to be appended to the ones we have already built, by previous calls to `AddVersion`.

For example:

```golang
builder := NewBuilder()
builder.AddVersion("1.2.3").AddVersion("4.2.3").Build()
```